### ISSUE TITLE
Project: Exploring a Data Repo

### ISSUE THREAD

**Issue by kadrlica** [LSSTScienceCollaborations/StackClub#90]
@drphilmarshall made a great start at this [here](https://github.com/LSSTScienceCollaborations/StackClub/blob/master/Basics/DataInventory.ipynb); however, I think we need to go a bit farther and document **what is in** each dataset rather than just listing the data sets.

**Comment by drphilmarshall** [comment]
@kadrlica is too kind: I'd say I made a _minimal_ start, rather than a
great one :-) Definitely agree with his assessment, though - when I ran out
of time I was starting to think about writing a `def` that:
* Displayed a big bold markdown section heading with the name of the dataset
* Printed the README of the dataset directory (or a warning if there is
none...)
* Listed the directory contents
* Computed some vital statistics (no of visits, sky are covered, no of
objects, no of populated columns in the object table, etc etc
* Plotted the sky coverage (that's what I needed Jim's DC2 notebook for)
and so on.
Then, the notebook could either loop over all datasets, and we'd have an
auto-inventory.

>

**Comment by drphilmarshall** [comment]
@rmorgan10 I just edited [my comment above](https://github.com/LSSTScienceCollaborations/StackClub/issues/90#issuecomment-412401628) to better reflect my current thinking about what the [Exploring A Data Repo](https://github.com/LSSTScienceCollaborations/StackClub/blob/project/data_inventory/drphilmarshall/Basics/Exploring_A_Data_Repo.ipynb) and [DataInventory](https://github.com/LSSTScienceCollaborations/StackClub/blob/project/data_inventory/drphilmarshall/Basics/DataInventory.ipynb) notebooks should each do, see what you think. These notes are consistent with the current state of the explorer notebook - I'm currently trying to figure out how to use the butler to check for the existence of various data types. 

Some more notes:
* In the DataInventory notebook I started trying to download an external notebook that contains functions to plot tracts and patches on the sky and import it as a module - I got close, but not close enough. I think the next step might be to just copy the relevant code from the remote notebook, and provide correct attribution to its author. All that should then happen in the Exploring notebook instead, while the Data Inventory notebook is kept clean. 
* Jeff Carlin showed some useful-looking visualization code in his tutorial stub on Friday, browse the project branches and commit history to find it.
* I think we can get quick code snippets for butler-querying repos from the DM team, either in #dm-newbies or #dm-pipelines - my guess is they have to do this kind of thing all the time.
* Gregory Dubois-Felsmann has comments on this project - he's part of the team developing the gen 3 butler, and said he wanted to talk to us! I'll point him at this thread so he can comment.
